server/interceptor/interceptor.go

```diff
@@ -158,7 +158,8 @@ func (tp *trillianProcessor) Before(ctx context.Context, req interface{}, method
 		return ctx, nil
 	}
 
-	ctx, spanEnd := spanFor(ctx, "Before")
+	// Don't want the Before to contain the action, so don't overwrite the ctx.
+	_, spanEnd := spanFor(ctx, "Before")
```

Comment:
Would this be better as `innerCtx, spanEnd := spanFor(ctx, "Before")` and then use `innerCtx` int he calls to `GetTree()` and `GetTokens()` below?

Reply:
yes, done - though it's a bit gnarly.